src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -1375,6 +1375,12 @@ bool CodeGen::genCreateAddrMode(GenTree*  addr,
 
     if (op2->IsIntCnsFitsInI32() && (op2->gtType != TYP_REF) && FitsIn<INT32>(cns + op2->AsIntConCommon()->IconValue()))
     {
+        // Don't build address modes out of non-foldable constants
+        if (!op2->AsIntConCommon()->ImmedValCanBeFolded(compiler, addr->OperGet()))
+        {
+            return false;
```

Comment:
Unrelated change?

Reply:
No, this is a necessary fix, otherwise we fold handle + const into an LEA on x86, and lose track of a reloc.

Reply:
I see; so it was an existing issue that was exposed by this change?

Reply:
Sort of ... looking at this again perhaps I should fix it differently.

VN-based constant prop won't propagate handles for 32 bit targets when `opts.compReloc` is true. But assertion-based constant prop doesn't have this restriction. And so with this change, the early flow opts enable an assertion-based constant prop of a handle, and on x86 constant handle is fodder for LEA formation.

So maybe I should update `optConstantAssertionProp` with similar restrictions as vn based const prop and revert the codegen change (or turn it into an assert).

Reply:
I think I will split this part of as a separate change, which should be zero-diff and will allow bit more refactoring than I'd do if I tried folding it in here. Should have that PR up soon.